### ISSUE TITLE
Error related to `xpath-iterator`

### ISSUE THREAD

**Issue by vemonet** [semantifyit/RocketRML#39]
Hi @sumutcan, I am getting an error related to `xpath-iterator` when trying to convert a CSV:

```
cached modules 14.2 MiB (javascript) 61.8 KiB (asset) 58.7 KiB (runtime) [cached] 1883 modules
javascript modules 637 KiB
  ./src/config/index.ts 4.86 KiB [built]
  ./src/config/rattScript.ts 6.32 KiB [built]
  ./node_modules/xpath-iterator/build/Release/xpathWrapper.node 626 KiB [built] [1 error]

ERROR in ./node_modules/xpath-iterator/build/Release/xpathWrapper.node 1:0
Module parse failed: Unexpected character '' (1:0)
You may need an appropriate loader to handle this file type, currently no loaders are configured to process this file. See https://webpack.js.org/concepts#loaders
(Source code omitted for this binary file)
 @ ./node_modules/xpath-iterator/index.js 1:14-58
 @ ./node_modules/rocketrml/src/input-parser/XmlParserCpp.js 4:18-43
 @ ./node_modules/rocketrml/src/input-parser/parser.js 10:21-46
 @ ./node_modules/rocketrml/src/index.js 5:15-47
 @ ./src/config/rattScript.ts 12:0-31 39:19-39
 @ ./src/config/index.ts 7:0-47 53:23-42
 @ ./src/containers/App/index.tsx 9:0-47 22:9-43 24:45-68 24:107-139 27:15-47 42:10-43 47:10-42 52:10-40
 @ ./src/index.tsx 5:0-35 10:76-79

webpack 5.74.0 compiled with 1 error in 515 ms
```

I use this code:

```typescript
import parser from "rocketrml";

const inputFiles = {"test_ldwizard.csv": "id,country,capital\n33,France,Paris\n32,Belgium,Brussels\n31,The Netherlands,Amsterdam"}
const options = {
    toRDF: true,
    verbose: true,
    xmlPerformanceMode: false,
    replace: false,
};
const result = await parser.parseFileLive(rmlMappings.toString(), inputFiles, options);
```

With these RML mappings:

```turtle
@prefix rml: <http://semweb.mmlab.be/ns/rml#>.
@prefix rr: <http://www.w3.org/ns/r2rml#>.
@prefix ql: <http://semweb.mmlab.be/ns/ql#>.
@prefix csvw: <http://www.w3.org/ns/csvw#>.
@prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>.
@prefix : <http://example.org/rules/>.
@prefix schema: <http://schema.org/>.
@prefix dbo: <http://dbpedia.org/ontology/>.

:TriplesMap rdf:type rr:TriplesMap;
    rml:logicalSource [
  rml:source [
  rdf:type csvw:Table;
  csvw:url "test_ldwizard.csv";
  csvw:dialect [
  rdf:type csvw:Dialect;
  csvw:delimiter ",";
  csvw:encoding "UTF-8"
]
];
  rml:referenceFormulation ql:CSV
];
    rr:subjectMap [ rr:template "https://data.netwerkdigitaalerfgoed.nl/e818d7/id/{id}" ];
    rr:predicateObjectMap [
  rr:predicate rdf:type;
  rr:objectMap [ rr:constant <http://www.w3.org/2000/01/rdf-schema#Resource> ]
], [
  rr:predicate <https://schema.org/countriesNotSupported>;
  rr:objectMap [ rml:reference "country" ]
], [
  rr:predicate <http://purl.org/dc/terms/modified>;
  rr:objectMap [ rml:reference "capital" ]
].
```

Note that I just installed `rocketrml` with `yarn add rocketrml`

It is defined as `"rocketrml": "^2.1.0",` in the `package.json` file, and it is effectively using `2.1.0` in `yarn.lock`

I am getting the same error when I deploy locally with node18, and when I deploy using node16 in a docker image

## Complete implementation

For more details, the implementation facing the error is here: https://github.com/vemonet/LDWizard/tree/add-rocketrml
You can easily reproduce it by cloning, going to the `add-rocketrml` branch and running:

```bash
yarn
yarn dev
```

Or deploy it with docker-compose in the `docker` folder:

```bash
cd docker
docker-compose up
```

The RML conversion is mainly defined here: https://github.com/vemonet/LDWizard/blob/add-rocketrml/src/config/rattScript.ts#L24

## Questions

Is it normal we are facing this error? Especially that we are not using any XML in our workflow (so RocketRML should only need to parse Turtle RDF and CSV)

Do we need to install an additional package? I tried installing the `xpath-iterator` package from the GitHub repo like it is done in RocketRML optional dependencies, but it does not fix the issue

**Comment by vemonet** [comment]
@sumutcan @ThibaultGerrier I did some digging: I managed to make it work by commenting out the import of the Cpp XML parser.

I am not an expert of the LDWizard inner workings but it seems like when the LDWizard builds with webpack it tries to resolve all defined imports, including XML Cpp, which fails. 

So maybe we could fix this by improving how the Cpp XML parser is imported, or not imported if the requirements are not installed.
